Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM perl:5.22
+
+ENV PERL_MM_USE_DEFAULT=1 PERL_CARTON_PATH=/carton
+
+COPY cpanfile cpanfile.snapshot /metacpan-web/
+WORKDIR /metacpan-web
+
+RUN cpanm App::cpm Carton && \
+    useradd -m metacpan-web -g users && \
+    mkdir /carton && \
+    cpm install -L /carton
+
+COPY . /metacpan-web
```

Comment:
Great, I'll remove the `COPY` then.  To add, I am also thinking that since Docker 17.05 has [multi-stage builds][0] now it would be possible to `COPY` the repo in much later, for production-style deployments.

[0]: https://docs.docker.com/engine/userguide/eng-image/multistage-build/

Just to note, the `CMD` can also be overridden at runtime by the user, e.g. via `docker run metacpan-api plackup -S starman` or similar.